luigi/contrib/hive.py

```diff
@@ -244,8 +248,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 def get_default_client():
-    if get_hive_syntax() == "apache":
+    syntax = get_hive_syntax()
+    if syntax == "apache":
         return ApacheHiveCommandClient()
+    elif syntax == "metastore":
```

Comment:
Can we introduce a new configuration that is `[hive]client` set as either `hivecli` or `thrift` (defaults to `hivecli`)?

This is using `[hive]release` which was intended to be the version of hive/hadoop (which translated to client differences for standard CLI hive). It is confusing to have a choice which is not a version. Also there may be some future thing that depends on us knowing the version outside of picking client.